### PR TITLE
Issue #1240 migrate credentials

### PR DESCRIPTION
**Pull request by elegoff**
Connected to https://github.com/slipstream/SlipStreamServer/issues/1240

### CODE REVIEW

File: ssclj/jar/test/com/sixsq/slipstream/ssclj/migrate/user_cred_test.clj [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+                                        :connector   "connector/exoscale-ch-gva"
+                                        :domain-name ""}
+                   :user               user1}
+        expected2 {:credentialTemplate {:href        "credential-template/store-cloud-cred-exoscale"
+                                        :key         "EXO0000000002"
+                                        :secret      "secret"
+                                        :connector   "connector/exoscale-ch-gva"
+                                        :domain-name ""}
+                   :user               user2}]
+    (is (= nil (t/extract-data nil nil nil)))
```
Comment (loomis) [inline ssclj/jar/test/com/sixsq/slipstream/ssclj/migrate/user_cred_test.clj]: Please use `are` to remove the repetition in this test.  Also you can replace `(= (nil ...))` with just `(nil? ...)`.

File: ssclj/jar/test/com/sixsq/slipstream/ssclj/migrate/user_cred_test.clj [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+                  :rules [{:type "ROLE", :principal "ADMIN", :right "ALL"}]}
+        expect-user1 {:owner {:principal "ADMIN", :type "ROLE"},
+                      :rules [{:type "ROLE", :principal "ADMIN", :right "ALL"}
+                              {:type "USER", :principal "user1", :right "VIEW"}
+                              ]}
+        expect-2users {:owner {:principal "ADMIN", :type "ROLE"},
+                       :rules [{:type "ROLE", :principal "ADMIN", :right "ALL"}
+                               {:type "USER", :principal "user1", :right "VIEW"}
+                               {:type "USER", :principal "user2", :right "VIEW"}]}]
+    (is (= base-acl (t/merge-acl nil)))
```
Comment (loomis) [inline ssclj/jar/test/com/sixsq/slipstream/ssclj/migrate/user_cred_test.clj]: Same here: group repetitive tests with an `are` statement.

File: ssclj/jar/test/com/sixsq/slipstream/ssclj/migrate/user_cred_test.clj [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+        base-expect {:credentialTemplate {:href        "credential-template/store-cloud-cred-exoscale",
+                                          :key         "aKey",
+                                          :secret      "aSecret",
+                                          :connector   "connector/exoscale-ch-gva",
+                                          :domain-name "",
+                                          :acl         {:owner {:principal "ADMIN", :type "ROLE"},
+                                                        :rules [{:type "ROLE", :principal "ADMIN", :right "ALL"}]}}}
+        add-user-rule (fn [x] (update-in base-expect [:credentialTemplate :acl :rules] conj {:type "USER" :principal (str "user/user" x) :right "VIEW"}))
+        expect-multi (map add-user-rule (range multiplicity))]
+    (is (thrown? AssertionError (t/generate-records nil)))
```
Comment (loomis) [inline ssclj/jar/test/com/sixsq/slipstream/ssclj/migrate/user_cred_test.clj]: Same here with the tests that throw exceptions.

File: ssclj/jar/build.boot [hunk 4]
```diff
@@ -268,4 +274,4 @@
          (comp
            (dev-env)
            (dev-fixture-env)
-           (repl)))
+           (repl)))
```
Comment (loomis) [inline ssclj/jar/build.boot]: Leave the newline.

File: ssclj/jar/src/com/sixsq/slipstream/ssclj/migrate/user_cred.clj [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+(def ^:const keys-cred-nuvlabox (vec (remove #{:domain-name :tenant-name} keys-cred-full)))
+(def ^:const keys-cred-stratuslab (vec (remove #{:domain-name :tenant-name} keys-cred-full)))
+(def ^:const keys-cred-otc keys-cred-full)
+(def ^:const keys-cred-openstack keys-cred-full)
+(def ^:const keys-cred-opennebula (vec (remove #{:domain-name :tenant-name} keys-cred-full)))
+(def ^:const keys-cred-exoscale (vec (remove #{:tenant-name} keys-cred-full)))
+(def ^:const keys-cred-ec2 (vec (remove #{:domain-name :tenant-name} keys-cred-full)))
+
+
+(def ^:const cat-nuvlabox #{"nuvlabox-arthur-harden", "nuvlabox-bertil-ohlin",
```
Comment (0xbase12) [inline ssclj/jar/src/com/sixsq/slipstream/ssclj/migrate/user_cred.clj]: We should move this hardcoded maps specific to nuvla in separated config edn file(s) and load them from there. It's too much hardcoded like that. We don't have time to do this for this release, so I will merge it like that but worth to change this next week

**Pull request merged**